Sources/Bitcoin/BitcoinUnspentSelector.swift

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2017-2018 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+
+enum BitcoinUnspentSelectorError: LocalizedError {
+    case insufficientFunds
+    case error(String)
+}
+
+public struct BitcoinUnspentSelector {
+    public let byteFee: Int64
+    public let dustThreshold: Int64
+
+    public init(byteFee: Int64 = 1, dustThreshold: Int64 = 3 * 182 + 1000) {
+        self.byteFee = byteFee
+        self.dustThreshold = dustThreshold
+    }
+
+    public func select(from utxos: [BitcoinUnspentTransaction], targetValue: Int64) throws -> (utxos: [BitcoinUnspentTransaction], fee: Int64) {
+        // if target value is zero, fee is zero
+        guard targetValue > 0 else {
+            return ([], 0)
+        }
+
+        // definitions for the following caluculation
+        let doubleTargetValue = targetValue * 2
+        var numOutputs = 2 // if allow multiple output, it will be changed.
+        var numInputs = 2
+        var fee: Int64 {
+            return calculateFee(nIn: numInputs, nOut: numOutputs)
```

Comment:
I find it confusing that this is a computed value. If it depends on numInputs and numOutputs it should be a function.